Can't build with nlohmann/json installed without `-DJSON_MultipleHeaders=ON`

**Does this issue affect the google-cloud-cpp project?** If the problem is with
the Google Cloud service exposed by the `google-cloud-cpp` libraries instead of
the client libraries themselves, you may consider opening a support request
instead. The `google-cloud-cpp` developers cannot help you troubleshoot problems
with the service itself.

Yes.

**What component of `google-cloud-cpp` is this related to?** For example, is
this related to bigtable (i.e., something in `google/cloud/bigtable`), or GCS
(i.e., something in `google/cloud/storage`)?

`google/cloud/internal/oauth2_universe_domain.h`

**Describe the bug** A clear and concise description of what the bug is.

https://github.com/googleapis/google-cloud-cpp/blob/f54ba19f2c4e6eaf64a54d09ff29de9bd85c07e5/google/cloud/internal/oauth2_universe_domain.h#L22 uses `nlohmann/json_fwd.hpp` not ` nlohmann/json.hpp` but `json_fwd.hpp` is available only when nlohmann/json is installed with `-DJSON_MultipleHeaders=ON`.

https://github.com/nlohmann/json?tab=readme-ov-file#integration

> You can further use file [include/nlohmann/json_fwd.hpp](https://github.com/nlohmann/json/blob/develop/include/nlohmann/json_fwd.hpp) for forward-declarations. The installation of json_fwd.hpp (as part of cmake's install step), can be achieved by setting `-DJSON_MultipleHeaders=ON`.

nlohmann/json provided by EPEL for RHEL 8 family doesn't provide `json_fwd.hpp`: https://packages.fedoraproject.org/pkgs/json/json-devel/epel-8.html#files

So we can't use the `json-devel` package on AlmaLinux 8. Here is an error log in apache/arrow:

https://github.com/ursacomputing/crossbow/actions/runs/9804176444/job/27071504390#step:8:3180

```text
-- stderr output is:
In file included from /root/rpmbuild/BUILD/apache-arrow-17.0.0.dev465/cpp/build/google_cloud_cpp_ep-prefix/src/google_cloud_cpp_ep/google/cloud/internal/oauth2_authorized_user_credentials.cc:16:
/root/rpmbuild/BUILD/apache-arrow-17.0.0.dev465/cpp/build/google_cloud_cpp_ep-prefix/src/google_cloud_cpp_ep/google/cloud/internal/oauth2_universe_domain.h:22:10: fatal error: nlohmann/json_fwd.hpp: No such file or directory
 #include <nlohmann/json_fwd.hpp>
          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

**To Reproduce** Steps to reproduce the behavior:

1. Prepare RHEL 8 based distribution such as AlmaLinux 8
1. Prepare EPEL by `sudo dnf install -y epel-release` or something
1. Install nlohmann/json by `sudo dnf install -y json-devel`
1. Build
1. See error

**Expected behavior** A clear and concise description of what you expected to
happen.

google-cloud-cpp can be built with nlohmann/json installed without `-DJSON_MultipleHeaders=ON`.

**Operating system:** If you are using a Linux distribution please include the
name and version of the distribution too.

AlmaLinux 8.

**What compiler and version are you using?** Please include the output of
`g++ -v` or `clang++ -v` or the equivalent command-line flag.

https://github.com/ursacomputing/crossbow/actions/runs/9804176444/job/27071504390#step:8:1579

```text
-- The C compiler identification is GNU 8.5.0
-- The CXX compiler identification is GNU 8.5.0
```

**What version of `google-cloud-cpp` are you using?** Please include the output
from `git rev-parse HEAD` if you are compiling from source, or the version
number from the applicable `google/cloud/*/version.h` file.

https://github.com/googleapis/google-cloud-cpp/releases/tag/v2.22.0

**Additional context** Add any other context about the problem here.

Can we use `nlohmann/json.h` instead for portability?

```diff
diff --git a/google/cloud/internal/oauth2_universe_domain.h b/google/cloud/internal/oauth2_universe_domain.h
index e74e438a0d..ca0fc6aaf8 100644
--- a/google/cloud/internal/oauth2_universe_domain.h
+++ b/google/cloud/internal/oauth2_universe_domain.h
@@ -19,7 +19,7 @@
 #include "google/cloud/status.h"
 #include "google/cloud/status_or.h"
 #include "google/cloud/version.h"
-#include <nlohmann/json_fwd.hpp>
+#include <nlohmann/json.hpp>
 #include <string>
 
 namespace google {
```

--- a/google/cloud/internal/oauth2_universe_domain.h
+++ b/google/cloud/internal/oauth2_universe_domain.h
@@ -19,7 +19,7 @@
 #include "google/cloud/status.h"
 #include "google/cloud/status_or.h"
 #include "google/cloud/version.h"
-#include <nlohmann/json_fwd.hpp>
+#include <nlohmann/json.hpp>
 #include <string>
 
 namespace google {

